packages/adapter/src/codec.ts

```diff
@@ -5,6 +5,26 @@ import JSONbigint from "json-bigint";
 
 export type DecodingError = string[];
 
+/**
+ * ```ts
+ * try {
+ *  myRuntimeEndpoint(myRequest);
+ * } catch (error){
+ *  if(error instanceOf UnexpectedRuntimeResponse) {
+ *    console.error("The Runtime answered an unexpected message, please contact our desk support",error)
+ *  } else  {
+ *    console.error("another type of error",error)
+ *  }
+ * }
+ * ```
+ */
+export class UnexpectedRuntimeResponse extends Error {
+  public type = "UnexpectedRuntimeResponse" as const;
+  constructor(message: string) {
+    super();
+  }
+}
+
```

Comment:
Did you forget to pass along message?

This is more ideomatic:
```ts
class UnexpectedRuntimeResponse extends Error {
  constructor(message?: string) {
    super(message);

    Object.setPrototypeOf(this, UnexpectedRuntimeResponse.prototype);
  }
}
```

Reply:
this something @hrajchert and I worked together on the side during a video call , I should remove it , Thx for spotting it and provided feedback on it !

Reply:
@hrajchert It does normally but `Error` breaks it so you have to manage it manually.